packages/react-devtools-shared/src/backend/renderer.js

```diff
@@ -1073,6 +1074,49 @@ export function attach(
     return null;
   }
 
+  function areHookInputsEqual(
+    nextDeps: Array<mixed>,
+    prevDeps: Array<mixed> | null,
+  ) {
+    if (prevDeps === null) {
+      return false;
+    }
+
+    for (let i = 0; i < prevDeps.length && i < nextDeps.length; i++) {
+      if (is(nextDeps[i], prevDeps[i])) {
+        continue;
+      }
+      return false;
+    }
+    return true;
+  }
+
+  function isEffect(memoizedState) {
+    return (
+      memoizedState !== null &&
+      typeof memoizedState === 'object' &&
+      memoizedState.hasOwnProperty('tag') &&
+      memoizedState.hasOwnProperty('create') &&
+      memoizedState.hasOwnProperty('destroy') &&
+      memoizedState.hasOwnProperty('deps') &&
+      (memoizedState.deps === null || Array.isArray(memoizedState.deps)) &&
+      memoizedState.hasOwnProperty('next')
+    );
+  }
```

Comment:
This function makes me nervous.

Reply:
Definitely. I haven't looked at where these data structures are created. Ideally we'd what kind of hook we're dealing with without checking it's memoized state.

Reply:
What happens if we rename the fields? Would some test fail, or would this code silently stop working?

Reply:
I think it should be possible to write a Profiler test that would catch this case. (Which we should do if we land this.)

Reply:
Actually a test already exists that would fail:
`ProfilingCache › should properly detect changed hooks`